### ISSUE TITLE
Error: unlinkat ./dump: device or resource busy

### ISSUE THREAD

**Issue by websolutions-hamburg** [ldez/traefik-certs-dumper#18]
If I use a `docker-compose.yml` file with the following configuration, I get this error message:

```
certdump_1  | Error: unlinkat ./dump: device or resource busy
certdump_1  | Usage:
certdump_1  |   traefik-certs-dumper file [flags]
certdump_1  | 
certdump_1  | Flags:
certdump_1  |   -h, --help            help for file
certdump_1  |       --source string   Path to 'acme.json' file. (default "./acme.json")
certdump_1  | 
certdump_1  | Global Flags:
certdump_1  |       --clean             Clean destination folder before dumping content. (default true)
certdump_1  |       --config string     config file (default is $HOME/.traefik-certs-dumper.yaml)
certdump_1  |       --crt-ext string    The file extension of the generated certificates. (default ".crt")
certdump_1  |       --crt-name string   The file name (without extension) of the generated certificates. (default "certificate")
certdump_1  |       --dest string       Path to store the dump content. (default "./dump")
certdump_1  |       --domain-subdir     Use domain as sub-directory.
certdump_1  |       --key-ext string    The file extension of the generated private keys. (default ".key")
certdump_1  |       --key-name string   The file name (without extension) of the generated private keys. (default "privatekey")
certdump_1  |       --watch             Enable watching changes.
certdump_1  | 
certdump_1  | 2019/04/25 11:38:40 root.go:46: unlinkat ./dump: device or resource busy
test_certdump_1 exited with code 1
```

```yml
version: "2"

services:
  certdump:
    image: ldez/traefik-certs-dumper:v2.2.0
    command: file --domain-subdir=true
    volumes:
      - ./acme.json:/acme.json:ro
      - ./letsencrypt:/dump
```

I fixed this with a new **not relative** dest path:
```yml
command: file --domain-subdir=true --dest /dump
```

This is my Docker version on a macOS 10.14.4:
```
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by websolutions-hamburg** [comment]
I'm not sure if it is the same problem.
The `clean` command will not work on the mapped folder. In my example the `./letsencrypt` folder is not cleaned up.

**Comment by ldez** [comment]
A workaround, could you try to use a sub-folder or set clean to false?

**Comment by websolutions-hamburg** [comment]
`--clean=false` works with the default `./dump` path.
  
And I've tried this without success:
```yml
command: file --domain-subdir
volumes:
  - ./acme.json:/acme.json:ro
  - ./letsencrypt/test:/dump
```
```yml
command: file --domain-subdir --dest=./dump/test
volumes:
  - ./acme.json:/acme.json:ro
  - ./letsencrypt/test:/dump/test
```

**Comment by ldez** [comment]
You have to use a sub-folder inside the mounted folder.

A fix come really soon.

**Comment by websolutions-hamburg** [comment]
Thanks for the soon fix!!!

And yes, this works (also with clean):
```yml
command: file --domain-subdir --dest=./dump/test --clean
volumes:
  - ./acme.json:/acme.json:ro
  - ./letsencrypt:/dump
```

**Comment by ldez** [comment]
It fixed in v2.2.1 :tada:

**Comment by ldez** [comment]
Due to linting error the v2.2.1 is not available, so I created a v2.2.2 :smile:
